skia-safe/src/core/typeface.rs

```diff
@@ -237,7 +237,19 @@ impl RCHandle<SkTypeface> {
             .if_true_then_some(|| name.as_str().into())
     }
 
-    // TODO: openStream()
+    pub fn to_font_data(&self) -> Option<(Vec<u8>, usize)> {
+        let mut ttc_index = 0;
+        StreamAsset::from_ptr(unsafe { sb::C_SkTypeface_openStream(self.native(), &mut ttc_index) })
+            .and_then(|mut stream| {
+                let length = unsafe { sb::C_SkStreamAsset_getLength(stream.native()) };
+                let mut data = vec![0u8; length];
+                let stream = stream.native_mut().as_stream_mut();
+                let read =
+                    unsafe { sb::C_SkStream_read(stream, data.as_mut_ptr() as _, data.len()) };
+                (read == data.len()).if_true_some((data, ttc_index.try_into().unwrap()))
```

Comment:
Would the index here always be zero? Should it be passed in?

Reply:
From the Skia documentation:
```
     *  Return a stream for the contents of the font data, or NULL on failure.
     *  If ttcIndex is not null, it is set to the TrueTypeCollection index
     *  of this typeface within the stream, or 0 if the stream is not a
     *  collection.
     *  The caller is responsible for deleting the stream.
```
It is returned only and indicates the index of the Typeface inside a true type collection, otherwise zero.